src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -686,5 +686,16 @@ public bool IsEquivalentTo(SyntaxToken token)
                 (Node == null && token.Node == null) ||
                 (Node != null && token.Node != null && Node.IsEquivalentTo(token.Node));
         }
+
+        /// <summary>
+        /// Returns true if these two tokens are considered "incrementally identical".  An incrementally identical token
+        /// occurs when a <see cref="SyntaxTree"/> is incrementally parsed using <see cref="SyntaxTree.WithChangedText"/>
+        /// and the incremental parser is able to take the token from the original tree and use it in its entirety in the
+        /// new tree.  In this case, the <see cref="SyntaxToken.ToFullString()"/> of each token will be the same, though 
+        /// will have different parents, and may occur at different positions in the respective trees.  If two tokens are
+        /// incrementally identical, all trivial of each node will be incrementally identical as well.
+        /// </summary>
```

Comment:
```suggestion
        /// </summary>
        /// <remarks>
        /// Incrementally identical tokens can also appear within the same syntax tree, or syntax trees that did not arise
        /// from <see cref="SyntaxTree.WithChangedText"/>.  This can happen as the parser is allowed to construct parse
        /// trees using shared tokens for efficiency.  In all these cases though, it will still remain true that the incrementally
        /// identical tokens will have different parents and may occur at different positions in their respective trees.
        /// </remarks>
```